core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -364,8 +364,16 @@ class BlockManagerMasterEndpoint(
         }
       }.getOrElse(Seq.empty)
 
+    val removeShuffleMergeFromShuffleServicesFutures =
```

Comment:
Do this only when push based shuffle is enabled to avoid the `getShufflePushMergerLocations` call.

Reply:
done